src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiIndexType.java

```diff
@@ -2,17 +2,49 @@
 
 import com.datastax.oss.driver.api.core.metadata.schema.IndexMetadata;
 import io.stargate.sgv2.jsonapi.exception.checked.UnsupportedCqlIndexException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 /** Defines what type of index configuration (vector or regular) */
 public enum ApiIndexType {
   // map, set, list
-  COLLECTION,
+  COLLECTION("collection"),
   // Something on a scalar column, and non analsed text index
-  REGULAR,
+  REGULAR("regular"),
   // Not available yet
-  TEXT_ANALYSED,
+  TEXT_ANALYSED("text-analysed"),
   // on a vector
-  VECTOR;
+  VECTOR("vector");
+
+  private final String indexTypeName;
+
+  private static final List<String> all =
```

Comment:
Naming: `static` class constants should be all upper-case, plus "all" is very generic (although it being `private` it's not part of API). So maybe name `ALL_TYPE_NAMES`?